lang/java/reef-common/src/main/java/org/apache/reef/util/MemoryUtils.java

```diff
@@ -96,6 +96,28 @@ public static void resetPeakUsage() {
     }
   }
 
+  /**
+   * Returns the total amount of physical memory on the current machine in megabytes.
+   *
+   * Some JVMs may not support the underlying API call.
+   *
+   * @return memory size in MB if the call succeeds; -1 otherwise
+   */
+  public static int getTotalPhysicalMemorySizeInMB() {
+
+    int memorySizeInMB;
+    try {
+      long memorySizeInBytes = ((com.sun.management.OperatingSystemMXBean) ManagementFactory
+                    .getOperatingSystemMXBean()).getTotalPhysicalMemorySize();
+
+      memorySizeInMB = (int) memorySizeInBytes / BYTES_IN_MEGABYTE;
```

Comment:
It is probably better to use `(int) (memorySizeInBytes / BYTES_IN_MEGABYTE)` to indicate that the cast is somewhat safe.